contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
-{% endhighlight %}
-* Reference theforeman/foreman as upstream:
-{% highlight bash %}
+```
+dnf module enable nodejs:14
+dnf install nodejs
+```
+
+#### Supported PostgreSQL versions
+
+In general, the latest version of PostgreSQL will work. Minimum version is currently 10, anything newer than that will do the job.
+
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf module enable postgresql:10
+dnf install postgresql-server
+```
+
+#### Libraries and database
+
+Some Ruby libraries (rubygems) require development tools (compilers, make), libraries and header files and Chrome driver to be present on the host. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf groupinstall "Development Tools"
+dnf install libvirt-devel postgresql-devel openssl-devel libxml2-devel sqlite-devel libxslt-devel zlib-devel readline-devel systemd-devel libcurl-devel krb5-devel
+dnf install chromedriver
+```
+
+Foreman requires PostgreSQL, even for development. SQLite is no longer supported and will not work. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+postgresql-setup --initdb
+```
+
+Configure security, for development purposes you can turn off authentication completely and let all users in on local sockets or loopback network interfaces. If you choose to use passwords, make sure to set them later:
+
+```
+grep trust /var/lib/pgsql/data/pg_hba.conf
+local   all             all                                     trust
+host    all             all             127.0.0.1/32            trust
+host    all             all             ::1/128                 trust
+```
+
+Start the service and create databases:
+
+```
+systemctl enable --now postgresql
+su postgres -s /bin/bash
+foreman# createuser -s foreman
+foreman# createdb --owner=foreman foreman
+foreman# createdb --owner=foreman foreman-test
+foreman# exit
+```
+
+#### Configure Foreman
+
+In the Foreman git repository, create and review necessary configuration files. If you named the databases "foreman" and "foreman-test" then no change is actually needed, but we do recommend to review configuration files because a change will be necessary depending on what you are planning to work on:
+
+```
+cp config/settings.yaml.example config/settings.yaml
+cp config/database.yml.example config/database.yml
+```
+
+Install the required Ruby gems. If you encounter a compilation error during rubygems installation, you are probably missing either development tools or library or headers. See above.
+
+```
+bundle install --path=vendor/
+```
+
+It is possible to skip some rubygems when you don't plan using specific features, although this is not recommended. You can probably use this if you cannot figure out what library are you missing:
+
+```
+bundle install --without="journald ec2 ovirt vmware openstack libvirt gce" --path=vendor/
+```
+
+Install the required Javascript modules:
+
+```
+npm install
```

Comment:
This uses the Ruby environment so `bundle exec` should be used.
```suggestion
bundle exec npm install
```

Reply:
No in this tutorial I suggest to use OS-level npm from Fedora module. This is actually correct.